pkg/cmd/checkendpoints/controller/backoff_recorder.go

```diff
@@ -0,0 +1,241 @@
+package controller
```

Comment:
Is it possible to consume the upstream library-go version of this? What changes would we need? This is a lot of code.

Reply:
oh, i was not aware it was on library-go, just copied over from CNO pki controller, thx!

Reply:
hmm, I guess you mean https://github.com/openshift/library-go/blob/master/pkg/operator/events/recorder_logging.go ?
That gives quite ugly logging?
e.g.

2020/11/10 18:27:40 reconciling (/v1, Kind=ServiceAccount) openshift-ovn-kubernetes/ovn-kubernetes-controller
W1110 18:27:40.982403  698742 recorder_logging.go:46] &Event{ObjectMeta:{dummy.1646359beaf46f13  dummy    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},InvolvedObject:ObjectReference{Kind:Pod,Namespace:dummy,Name:dummy,UID:,APIVersion:v1,ResourceVersion:,FieldPath:,},Reason:EndpointDetectionFailure,Message:PodNetworkConnectivityCheck.controlplane.operator.openshift.io/network-check-tester-rcarrillocruz-diag-6-dstf9-worker-b-pgd98-to-openshift-apiserver-service-cluster -n openshift-apiserver: the namespace of the provided object does not match the namespace sent on the request,Source:EventSource{Component:,Host:,},FirstTimestamp:2020-11-10 18:27:40.982251283 +0100 CET m=+27.659866000,LastTimestamp:2020-11-10 18:27:40.982251283 +0100 CET m=+27.659866000,Count:1,Type:Warning,EventTime:0001-01-01 00:00:00 +0000 UTC,Series:nil,Action:,Related:nil,ReportingController:,ReportingInstance:,}
2020/11/10 18:27:41 update was successful

Reply:
ended up using the LoggingRecorder from pki, making its own package